python/idsse_common/idsse/common/vectaster.py

```diff
@@ -0,0 +1,397 @@
+
+    Args:
+        pnt1 (Tuple[float, float]): One of the line segment end points in x,y
+        pnt2 (Tuple[float, float]): The other line segment end point in x,y
+        exclude_first (bool): If true, pnt1 is not included in returned list
+
+    Returns:
+        List[Pixel]: List of x,y tuples in pixel space
+    """
+    x1, y1 = pnt1
+    x2, y2 = pnt2
+    dx = x2 - x1
+    dy = y2 - y1
+
+    pixels = []
+    if not exclude_first:
+        pixels.append((int(x1), int(y1)))
+
+    slope = float(dy) / float(dx) if dx else 0
+    if dx != 0 and -1 <= slope <= 1:
+        intercept = y1 - slope * x1
+        dx = 1 if x2 > x1 else -1
+        while x1 != x2:
+            x1 += dx
+            y1 = round_(slope * x1 + intercept)
+            pixels.append((int(x1), y1))
+    else:
+        slope = float(dx) / float(dy) if dy else 0
+        intercept = x1 - slope * y1
+        dy = 1 if y2 > y1 else -1
+        while y1 != y2:
+            y1 += dy
+            x1 = round_(slope * y1 + intercept)
+            pixels.append((x1, int(y1)))
+
+    return pixels
+
+
+# pylint: disable=too-many-locals
+def _pixels_for_polygon(
+        polygon_boundary: LinearRing
+) -> List[Pixel]:
+    """Get all pixels in a polygon using a line scan algorithm
+
+    Args:
+        polygon_boundary (LineString): The outside of a polygon as linestring, must
+                                       repeat first vertex as last, likely generated by
+                                       call the polygon.boundary property
+    Returns:
+        List[Pixel]: List of x,y tuples in pixel space
+    """
+    xmin, ymin, _, ymax = polygon_boundary.bounds
+    x_offset = 0 if xmin >= 0 else int(1-xmin)
+    ymin = floor(ymin)
+    ymax = floor(ymax)
+
+    edge_info = []
+    for (x1, y1), (x2, y2) in zip(polygon_boundary.coords[:-1], polygon_boundary.coords[1:]):
+        int_y1, int_y2 = int(y1), int(y2)
+        if int_y1 != int_y2:  # only need to include if y values differ
+            if int_y1 > int_y2:  # need to swap order so y1 < y2
+                x1, x2 = x2, x1
+                y1, y2 = y2, y1
+                int_y1, int_y2 = int_y2, int_y1
+
+            edge_info.append((int_y1, int_y2, x1+x_offset, (x2 - x1)/(y2 - y1)))
+
+    pixels = []
+    for y in range(ymin, ymax+1):
+        x_list = []
+        for y1, y2, x, slope in edge_info:
+            if y1 <= y <= y2:
+                x_list.append(int(x + (y-y1) * slope))
+
+        x_list.sort()
+
+        for x1, x2 in zip(x_list[:-1], x_list[1:]):
+            pixels.extend([(x, y) for x in range(x1-x_offset, x2-x_offset+1)])
+
+    return pixels
+
+
+def _round_x_y(x, y, rounding: Union[str, RoundingMethod]) -> Pixel:
+    if isinstance(rounding, str):
+        rounding = RoundingMethod[rounding.upper()]
+
+    if rounding is RoundingMethod.ROUND:
+        return (round_(x), round_(y))
+    if rounding is RoundingMethod.FLOOR:
+        return (floor(x), floor(y))
+    return (int(x), int(y))
```

Comment:
I wrote myself an identical private method in `GridProj._round_pixel()` (on [line 313](https://github.com/NOAA-GSL/idss-engine-commons/blob/main/python/idsse_common/idsse/common/grid_proj.py#L313)). We could pull this function out into `utils.py` and have both modules import it from there?

Reply:
I copied if from grid_proj, so agree we should move it into utils.py